eclair-core/src/main/scala/fr/acinq/eclair/swap/SwapInSender.scala

```diff
@@ -0,0 +1,289 @@
+       (claim_by_csv) [ClaimSwapCsv] |                                |
+
+  */
+
+  def apply(nodeParams: NodeParams, watcher: ActorRef[ZmqWatcher.Command], register: actor.ActorRef, wallet: OnChainWallet): Behavior[SwapCommands.SwapCommand] =
+    Behaviors.setup { context =>
+      Behaviors.receiveMessagePartial {
+        case StartSwapInSender(replyTo, amount, swapId, shortChannelId, channelId) =>
+          val actor = new SwapInSender(replyTo, amount, swapId, shortChannelId, channelId, nodeParams, watcher, register, wallet, context)
+          actor.CreateSwap()
+        case RestoreSwapInSender(replyTo, d) =>
+          ShortChannelId.fromCoordinates(d.request.scid) match {
+            case Success(shortChannelId) => new SwapInSender(replyTo, d.request.amount.sat, d.request.swapId, shortChannelId, d.channelId, nodeParams, watcher, register, wallet, context)
+                .AwaitClaimPayment(d.request, d.agreement, d.invoice, d.openingTxBroadcasted)
+            case Failure(e) => replyTo ! LocalError(d.request.swapId, e)
+              Behaviors.stopped
+          }
+        case AbortSwapInSender => Behaviors.stopped
+      }
+    }
+}
+
+private class SwapInSender(replyTo: ActorRef[SwapResponses.Response], amount: Satoshi, swapId: String, implicit val shortChannelId: ShortChannelId, channelId: ByteVector32, nodeParams: NodeParams, implicit val watcher: ActorRef[ZmqWatcher.Command], implicit val register: actor.ActorRef, implicit val wallet: OnChainWallet, implicit val context: ActorContext[SwapCommands.SwapCommand]) {
+  val protocolVersion = 1
+  val noAsset = ""
+  private val invoiceAdapter = context.messageAdapter[Bolt11Invoice](InvoiceResponse)
+  private val openingConfirmedAdapter = context.messageAdapter[ZmqWatcher.WatchTxConfirmedTriggered](OpeningConfirmed)
+  private val claimByCoopConfirmedAdapter = context.messageAdapter[ZmqWatcher.WatchTxConfirmedTriggered](ClaimTxConfirmed)
+  private val csvDelayConfirmedAdapter = context.messageAdapter[ZmqWatcher.WatchTxConfirmedTriggered](CsvDelayConfirmed)
+  private val claimByCsvConfirmedAdapter = context.messageAdapter[ZmqWatcher.WatchTxConfirmedTriggered](ClaimTxConfirmed)
+  implicit val timeout: Timeout = 30 seconds
+  private val keyManager: SwapKeyManager = nodeParams.swapKeyManager
+  private implicit val feeRatePerKw: FeeratePerKw = nodeParams.onChainFeeConf.feeEstimator.getFeeratePerKw(target = nodeParams.onChainFeeConf.feeTargets.fundingBlockTarget)
+  private val maxPremium = (feeRatePerKw * claimByInvoiceTxWeight / 1000).toLong // TODO: how should swap sender calculate an acceptable premium?
+
+  private def makerPrivkey(): PrivateKey = keyManager.openingPrivateKey(SwapKeyManager.keyPath(swapId)).privateKey
+  private def makerPubkey(): PublicKey = makerPrivkey().publicKey
+  private def takerPubkey(agreement: SwapInAgreement): PublicKey = PublicKey(ByteVector.fromValidHex(agreement.pubkey))
+
+  private def CreateSwap(): Behavior[SwapCommand] = {
```

Comment:
nit: same as above, should start with a lowercase letter

Reply:
fixed in [03323a3](https://github.com/ACINQ/eclair/pull/2342/commits/03323a3699644757ed366f21339c8ad86ad699fe)